cloud-storage.go

```diff
@@ -38,6 +38,38 @@ func NewCloudStorage(
 	awsS3AccessKeyID string,
 	awsS3SecretAccessKey string,
 
+	gcpCredentialsJSON string,
+	gcpStorageEmulatorHost string,
+) (CloudStorage, error) {
+	return NewCloudStorageV2(
+		ctx,
+		isTesting,
+		bucketProvider,
+		bucketName,
+
+		awsS3Endpoint,
+		awsS3Region,
+		awsS3AccessKeyID,
+		awsS3SecretAccessKey,
+		false,
+		gcpCredentialsJSON,
+		gcpStorageEmulatorHost,
+	)
+}
+
+//nolint:funlen
+func NewCloudStorageV2(
```

Comment:
can we use a new options map including other potential arguments, otherwise, it will introduce v2, v3, ...